src/django_browser_reload/views.py

```diff
@@ -145,19 +150,40 @@ def events(request: HttpRequest) -> HttpResponseBase:
     if not request.accepts("text/event-stream"):
         return HttpResponse(status=HTTPStatus.NOT_ACCEPTABLE)
 
-    def event_stream() -> Generator[bytes, None, None]:
-        while True:
-            yield message("ping", versionId=version_id)
-
-            should_reload = should_reload_event.wait(timeout=PING_DELAY)
-            if should_reload:
-                should_reload_event.clear()
-                yield message("reload")
-
-    response = StreamingHttpResponse(
-        event_stream(),
-        content_type="text/event-stream",
-    )
+    if isinstance(request, ASGIRequest):
+        if DJANGO_VERSION < (4, 2):
+            err_msg = "We cannot support hot reload with ASGI for Django < 4.2"
+            warnings.warn(err_msg, stacklevel=2)
```

Comment:
not sure a Python warning is helpful, leaving a message in the browser console seems more logical

Reply:
After some playing around, leaving a browser console message would be unnecessarily complicated. I'm only going to put it in the HTTP response content, which users can debug if necessary. The ASGI support is documented as Django 4.2+ so hopefully no one will be surprised when it doesn't work on older versions.